readthedocs/projects/views/private.py

```diff
@@ -111,38 +111,53 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class ProjectUpdate(ProjectSpamMixin, PrivateViewMixin, UpdateView):
+class ProjectMixin(PrivateViewMixin):
 
-    form_class = UpdateProjectForm
     model = Project
-    success_message = _('Project settings updated')
-    template_name = 'projects/project_edit.html'
     lookup_url_kwarg = 'project_slug'
     lookup_field = 'slug'
+    context_object_name = 'project'
 
     def get_queryset(self):
         return self.model.objects.for_admin_user(self.request.user)
 
+
+class ProjectUpdate(ProjectSpamMixin, ProjectMixin, UpdateView):
+
+    form_class = UpdateProjectForm
+    success_message = _('Project settings updated')
+    template_name = 'projects/project_edit.html'
+
     def get_success_url(self):
         return reverse('projects_detail', args=[self.object.slug])
 
 
-class ProjectAdvancedUpdate(ProjectSpamMixin, PrivateViewMixin, UpdateView):
+class ProjectAdvancedUpdate(ProjectSpamMixin, ProjectMixin, UpdateView):
 
     form_class = ProjectAdvancedForm
-    model = Project
     success_message = _('Project settings updated')
     template_name = 'projects/project_advanced.html'
-    lookup_url_kwarg = 'project_slug'
-    lookup_field = 'slug'
-
-    def get_queryset(self):
-        return self.model.objects.for_admin_user(self.request.user)
 
     def get_success_url(self):
         return reverse('projects_detail', args=[self.object.slug])
 
 
+class ProjectDelete(ProjectMixin, DeleteView):
+
+    success_message = _('Project deleted')
+    template_name = 'projects/project_delete.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['is_superproject'] = (
```

Comment:
It seems the function-based view has the `project` injected in the context as well. I'm not sure if it's required, or it just comes from `context_object_name` (attribute class in the mixin).

Reply:
Yeah, it comes from the `context_object_name`